tools/crdb-physical-replication.sh

```diff
@@ -0,0 +1,358 @@
+    local enabled=$(cockroach sql --execute "$sql" --url "$url" --certs-dir "$certsdir" --format 'tsv' | tail -1 | xargs)
+
+    if [[ $enabled != 't' ]];
+    then
+        echo 'Enabling physical replication...'
+        sql='SET CLUSTER SETTING cross_cluster_replication.enabled = true;'
+        cockroach sql --execute "$sql" --certs-dir "$certsdir" --url "$url"
+    else
+        echo "Physical replication is enabled"
+    fi
+
+    if [[ $cluster == 'primary' ]];
+    then
+        sql='SHOW CLUSTER SETTING kv.rangefeed.enabled;'
+        enabled=$(cockroach sql --execute "$sql" --url "$url" --certs-dir "$certsdir" --format 'tsv' | tail -1 | xargs)
+
+        if [[ $enabled != 't' ]];
+        then
+            echo 'Enabling rangefeeds...'
+            sql='SET CLUSTER SETTING kv.rangefeed.enabled = true;'
+            cockroach sql --execute "$sql" --certs-dir "$certsdir" --url "$url"
+        else
+            echo "Rangefeeds enabled"
+        fi
+
+        echo "Checking replication user..."
+
+        local username='replication'
+        local sql="select username from [show users] where username='$username';"
+        local users=$(cockroach sql --execute "$sql" --url "$url" --certs-dir "$certsdir" --format 'tsv')
+        
+        if [[ $users == *"$username"* ]]; then
+            echo "User '$username' already exists. Hopefully you have saved the password."
+            echo ''
+            echo "${bold}Connection string for later use by the Standby cluster:${normal}"
+            local urlforstandby="postgresql://replication:<the password>@$listenaddr/?options=-ccluster=system&sslmode=verify-full&sslinline=true&sslrootcert=$cacrtpath"
+            echo $urlforstandby
+            echo ''
+        else
+            # pwgen is not on all systems, is there a better choice?
+            # https://unix.stackexchange.com/questions/230673/how-to-generate-a-random-string
+            pw=$(openssl rand -base64 20)
+            echo "Creating user '$username' password..."
+            sql="CREATE USER $username WITH LOGIN PASSWORD '$pw';"
+            cockroach sql --execute "$sql" --url "$url" --certs-dir "$certsdir"
+
+            echo "Granting REPLICATION role..."
+            cockroach sql --execute "GRANT SYSTEM REPLICATION TO $username;" --url "$url" --certs-dir "$certsdir"
+
+            echo "${bold}Remember this connection string for later use by the Standby cluster:${normal}"
+            local urlforstandby="postgresql://$username:$pw@$listenaddr/?options=-ccluster=system&sslmode=verify-full&sslrootcert=$cacrtpath"
+            echo $urlforstandby
+            echo ''
+        fi
+    fi
+}
+
+if [[ $ps == "P" || $ps == "p" ]];
+then
+    label='Primary'
+    cluster='primary'
+    configprofile='replication-source'
+    listenaddr='localhost:9001'
+    httpaddr='localhost:8081'
+
+    intro
+    certs
+    server
+
+    url="postgresql://root@$listenaddr/?options=-ccluster=system&sslmode=verify-full"
```

Comment:
Are you sure you want the workload to run against the system tenant?

Reply:
Heh, no, probably not.